### ISSUE TITLE
Making propify infectious again

### ISSUE THREAD

**Issue by pygy** [lhorie/mithril.js#904]
This is related to #903 

A common pattern in Mithril

```JS
component = {
    controller: function () {
        this.transmogrified = m.request(...).then(transmogrify)
    }
}
```

If propify isn't infectious that pattern breaks with non-mithril promises, you end up with a normal promise instead.

There are other, more complex patterns that are also broken currently, like

```
        this.fooBarized = this.transmogrified.then(fooBar)
```

Infectious propification doesn't make a promise non-compliant, it's a `prop` behavior mixin. The promise is still immutable (unlike what v0.2.1 does IIUC), and behaves as expected by the A+ spec (if it was compliant in the first place).

**Comment by isiahmeadows** [comment]
@pygy I'm working on this now, and damn is it hard to do. I've combined the logic for Mithril deferreds and ES6 promises, but that only is saving on typing. Let's just say the magical async-to-sync access is not trivial to pull off with ES6 promises. I've already pulled the es6-promise polyfill as a new dependency, but my biggest struggle is dealing with the shitty, very non-obvious race condition causing Promise values not to be passed down the chain. I'm surprised the old version worked at all. :angry:
